# V Bucle `for` en Python

# Contenido 

- [1. Introduccion](#Introduccion)
- [2. Bucle `for`: intro](#Bucle-for-intro)
- [3. Bucle `for`: analisis](#Bucle-for-analisis)
- [4. Estructura del bucle `for`](#Estructura-bucle-for)
- [5. Abrir un archivo](#Abrir-un-archivo)
- [6. Calcular promedio](#Calcular-promedio)
- [7. Funcion `append()` para calcular un promedio](#Append-calcular-promedio)

<a id="Introduccion"></a>
# 1. Instroducción 
[[Contenido]](#Contenido)

En esta lección aprenderemos cómo usar un bucle `for` para repetir un proceso y analizar un conjunto de datos, y cómo leer un conjunto de datos grande como una lista de listas.

En ejercicios anteriores cuando trabajábamos con cinco filas de datos y queríamos calcular el promedio de alguna columna de estos datos lo que hacíamos era leer el dato de cada fila y almacenarlo en una variable, posteriormente sumabamos estas cinco variables y la suma la dividiamos entre cinco para calcular el promedio.

Ahora imaginemos que el conjunto de datos no sólo tiene cinco filas sino miles de ellas, el procedimiento anterior sería bastante complejo, ya que almacenar cada valor en una variable tomaría demasiado tiempo con lo que esto sería impractico. Con esta observacion podemos concluir que necesitamos una mejor manera de llevar a cabo este tipo de  procesos repetitivos.

Afortunadamente Python nos ofrece una forma fácil de repetir procesos, el bucle `for`, el cual nos ayuda a repetir procesos cientos, miles o incluso millones de veces.

<a id="Bucle-for-intro"></a>
## 2. Bucle `for`: intro
[[Contenido]](#Contenido) 
[[Ejercicio de esta sección]](#Instrucciones-Bucle-for-intro)

Aprendamos a usar el bucle `for` para repetir un proceso varias veces. Supongamos que tenemos la lista `app_ratings = [3.5, 4.5, 4.5, 4.5, 4.5]` y el proceso que queremos llevar a cabo es: imprimir cada uno de los elementos de la lista `app_ratings`.  A continuación se muestra cómo podemos implementar este proceso utilizando el bucle `for` en Python.

In [ ]:
app_ratings = [3.5, 4.5, 4.5, 4.5, 4.5]

for element in app_ratings:
    print(element)

A esta técnica la llamamos **bucle** porque recorremos cada elemento de la lista y realizamos una acción con cada elemento. Debido a que siempre comenzamos con `for` (como en `for element in app_ratings:`), llamamos a esta técnica **bucle for**.

<a id="Instrucciones-Bucle-for-intro"></a>
## Instrucciones 
[[Contenido]](#Contenido) 
[[Teoria de esta sección]](#Bucle-for-intro)

Utilice la nueva técnica que has aprendido para imprimir todos los nombres de aplicaciones en la lista `app_names` la cual se encuentra en la celda de abajo.

Escribe tu codigo abajo de esta sección ⏬

---

In [ ]:
app_names = ['Facebook', 'Instagram', 'Clash of Clans', 'Fruit Ninja Classic', 'Minecraft: Pocket Edition']

# Ingresa aqui debajo


Escribe tu codigo arriba de esta sección ⏫

---

<a id="Bucle-for-analisis"></a>
## 3. Bucle `for`: analisis
[[Contenido]](#Contenido) 
[[Ejercicio de esta sección]](#Instrucciones-Bucle-for-analisis)

En la celda de abajo se muestra el método para calcular la calificación promedio de las aplicacións (promedio de 5 aplicaciones una por cada fila).

In [ ]:
row_1 = ['Facebook', 0.0, 'USD', 2974676, 3.5]
row_2 = ['Instagram', 0.0, 'USD', 2161558, 4.5]
row_3 = ['Clash of Clans', 0.0, 'USD', 2130805, 4.5]
row_4 = ['Fruit Ninja Classic', 1.99, 'USD', 698516, 4.5]
row_5 = ['Minecraft: Pocket Edition', 6.99, 'USD', 522012, 4.5]

app_data_set = [row_1, row_2, row_3, row_4, row_5]
avg_rating = (
              app_data_set[0][-1] + # Ultimo elemento de la fila 1 (row_1)
              app_data_set[1][-1] + # Ultimo elemento de la fila 2 (row_2)
              app_data_set[2][-1] + # Ultimo elemento de la fila 3 (row_3)
              app_data_set[3][-1] + # Ultimo elemento de la fila 4 (row_4)
              app_data_set[4][-1]   # Ultimo elemento de la fila 5 (row_5)
             ) / 5

print(avg_rating)

Afortunadamente, podemos usar un bucle `for` para calcular la calificación promedio de todas las aplicaciones. **¿Pero por donde empezamos?**. El primer paso es determinar el proceso que queremos utilizar para extraer cada calificación individual del conjunto de datos. El proceso es: **"para cada fila en** `app_data_set` **extraer el último elemento"**. En la celda de abajo se muestra cómo podemos traducir ese proceso a la sintaxis de Python e imprimir el resultado de cada iteración:

In [ ]:
row_1 = ['Facebook', 0.0, 'USD', 2974676, 3.5]
row_2 = ['Instagram', 0.0, 'USD', 2161558, 4.5]
row_3 = ['Clash of Clans', 0.0, 'USD', 2130805, 4.5]
row_4 = ['Fruit Ninja Classic', 1.99, 'USD', 698516, 4.5]
row_5 = ['Minecraft: Pocket Edition', 6.99, 'USD', 522012, 4.5]

app_data_set = [row_1, row_2, row_3, row_4, row_5]

for row in app_data_set:
    rating = row[-1]
    print(rating)

El siguiente paso para calcular la calificación promedio de todas las aplicaciones es calcular la suma y almacenar el resultado. Aprenderemos cómo hacer esto mas adelante, pero primero es import que entendamos lo que sucede en el bucle `for` que aparece en al celda anterior. 

El bulce `for`  aísla cada fila de `app_data_set` una a la vez y la asigna a la variable `row`. Veamos un ejemplo de cómo podemos usar un bucle `for` para imprimir el contenido de cada fila del conjunto de datos `app_data_set`:

In [ ]:
app_data_set = [row_1, row_2, row_3, row_4, row_5]

for row in app_data_set:
    print(row)

Utilizando el bucle `for`, con solo dos líneas de código, imprimimos el contenido de las cinco listas del conjunto de datos `app_data_set`. Ahora comparemoslo con el siguiente enfoque que no utiliza el bucle `for` pero que arroja resultados equivalentes:

In [ ]:
app_data_set = [row_1, row_2, row_3, row_4, row_5]

print(app_data_set[0])
print(app_data_set[1])
print(app_data_set[2])
print(app_data_set[3])
print(app_data_set[4])

Usar la técnica en la celda anterior requiere que escribamos una línea de código para cada fila del conjunto de datos. Miestras que usando la técnica `for row in app_data_set` requiere que escribamos solo dos líneas de código, independientemente del número de filas en el conjunto de datos, el cual puede llegar incluso a ser de millones.

Ahora mismo uno de nuestros objetivos es utilizar la tecnica del bucle `for` para calcular la calificación promedio de las cinco aplicaciones. Un segundo objetivo es calcular la calificación promedio de un conjunto de datos mas grande. 

Un punto importante en la estructura de un ciclo `for` es el sangrado del codigo, ya que debemos **sangrar el código** que queremos que se repita cuatro caracteres de espacio a la derecha. Observa la sangría antes `rating` y `print` en la siguiente celda.

In [ ]:
app_data_set = [row_1, row_2, row_3, row_4, row_5]

for row in app_data_set:
    rating = row[-1]
    print(rating)

Técnicamente, solo necesitamos sangrar el código al menos un carácter de espacio a la derecha, pero la convención en la comunidad de Python es usar cuatro caracteres de espacio. Esto ayuda con la legibilidad; es decir, es más fácil para otras personas leer tu código y será más fácil para ti leer el de ellos.

<a id="Instrucciones-Bucle-for-analisis"></a>
## Instrucciones
[[Contenido]](#Contenido) 
[[Teoria de esta sección]](#Bucle-for-analisis)

Utilice la técnica del ciclo `for` que has aprendido para imprimir todos los recuentos de calificaciones de aplicaciones en la lista de listas `app_data_set`. La lista de listas `app_data_set` se encuntra en la celda de abajo.

- Recorre (itera) la lista de listas `app_data_set`. Para cada una de las cinco iteraciones del bucle (`for` cada fila `in app_data_set`):
    - Extrae el recuento de calificaciones de la aplicación y guárdalo en una variable denominada `rating_count`. El recuento de calificaciones es el cuarto elemento de cada fila (p.ej. `row_3[3]`).
    - Imprime `rating_count`.

Escribe tu codigo abajo de esta sección ⏬

---

In [ ]:
row_1 = ['Facebook', 0.0, 'USD', 2974676, 3.5]
row_2 = ['Instagram', 0.0, 'USD', 2161558, 4.5]
row_3 = ['Clash of Clans', 0.0, 'USD', 2130805, 4.5]
row_4 = ['Fruit Ninja Classic', 1.99, 'USD', 698516, 4.5]
row_5 = ['Minecraft: Pocket Edition', 6.99, 'USD', 522012, 4.5]

app_data_set = [row_1, row_2, row_3, row_4, row_5]

# Escribe tu codigo debajo



Escribe tu codigo arriba de esta sección ⏫

---

<a id="Estructura-bucle-for"></a>
## 4. Estructura del bucle `for`
[[Contenido]](#Contenido) 
[[Ejercicio 1 de esta sección]](#Instrucciones-Estructura-bucle-for-1)
[[Ejercicio 2 de esta sección]](#Instrucciones-Estructura-bucle-for-2)

Hasta ahora, hemos aprendido cómo usar bucles `for` para imprimir elementos de una lista. Ahora, veamos cómo usar bucles `for` para sumar valores de una lista. Para hacer esto, necesitamos comprender la estructura de un bucle `for`. 

In [ ]:
una_lista = [1, 2, 3]

for valor in una_lista:
    print(valor)
    print(valor * 2)

El código sangrado debajo del bucle `for` (el "**cuerpo**" del bucle) se ejecuta una vez para cada elemento de la lista. Entonces, si la lista tiene tres elementos, el código en el cuerpo del bucle se ejecuta tres veces. A cada ejecución del cuerpo la llamamos "**iteración**".

Durante cada iteración, la "**variable de iteración**" tomará el valor del siguiente elemento de la lista. Por ejemplo, si la lista es `[1, 3, 5]`, entonces durante la primera iteración la variable de iteración será `1`, durante la segunda iteración será `3` y durante la tercera iteración será `5`.

El código fuera del cuerpo del bucle puede interactuar con el código dentro del cuerpo del bucle. Por ejemplo, en el siguiente código, hacemos lo siguiente:

- Inicializamos una variable `a_sum` con un valor de cero fuera del cuerpo del bucle.
- Hacemos un bucle (o iteramos) sobre `a_list`. Para cada iteración del ciclo, hacemos lo siguiente:
    - Realizamos una suma (dentro del cuerpo del bucle) entre el valor actual de la variable de iteración `value` y el valor actual almacenado en `a_sum` (`a_sum` se definió fuera del cuerpo del bucle).
    - Asignamos el resultado de la suma a `a_sum` (dentro del cuerpo del bucle).
    - Imprimimos el valor de la variable `a_sum` (dentro del cuerpo del bucle). Observa que el valor de `a_sum` cambia después de cada suma. Al final del ciclo, `a_sum` tiene el valor `9`, que equivale a la suma de los números en `a_list` `(1 + 3 + 5)`.

In [ ]:
a_list = [1, 3, 5]

a_sum = 0
for value in a_list:
    a_sum = a_sum + value
    print(a_sum)

print(a_sum)

En la celda de anterior, creamos una forma de sumar los números en una lista. Podemos utilizar esta técnica para sumar la cantidad de calificaciones de aplicaciones en nuestro conjunto de datos.

<a id="Instrucciones-Estructura-bucle-for-1"></a>
## Instrucciones parte 1
[[Contenido]](#Contenido) 
[[Teoria de esta sección]](#Estructura-bucle-for)
[[Ejercicio 2 de esta sección]](#Instrucciones-Estructura-bucle-for-2)

Calcula la cantidad total de calificaciones de aplicaciones para las aplicaciones almacenadas en la lista `app_data_set` que aparece en la celda de abajo.

- Inicializa una variable denominada `rating_sum` con un valor de cero fuera del cuerpo del bucle.
- Recorre (itera) la lista de listas `app_data_set`. Para cada una de las cinco iteraciones del bucle (para cada fila en `app_data_set`):
    - Obten la cantidad de calificaciones (posición 3 de cada fila) para esa aplicación y guárdala en una variable llamada `rating_count`.
    - Agrega el valor almacenado en `rating_count` al valor actual de `rating_sum` y asigna el resultado nuevamente a `rating_sum`.
    - Imprime `rating_sum`.

Escribe tu codigo abajo de esta sección ⏬

---

In [ ]:
row_1 = ['Facebook', 0.0, 'USD', 2974676, 3.5]
row_2 = ['Instagram', 0.0, 'USD', 2161558, 4.5]
row_3 = ['Clash of Clans', 0.0, 'USD', 2130805, 4.5]
row_4 = ['Fruit Ninja Classic', 1.99, 'USD', 698516, 4.5]
row_5 = ['Minecraft: Pocket Edition', 6.99, 'USD', 522012, 4.5]

app_data_set = [row_1, row_2, row_3, row_4, row_5]

# Escribe tu codigo debajo



Escribe tu codigo arriba de esta sección ⏫

---

<a id="Instrucciones-Estructura-bucle-for-2"></a>
## Instrucciones parte 2
[[Contenido]](#Contenido) 
[[Teoria de esta sección]](#Estructura-bucle-for)
[[Ejercicio 1 de esta sección]](#Instrucciones-Estructura-bucle-for-1)

Ahora que sabemos cómo iterar sobre una lista y almacenar los resultados, tenemos las herramientas que necesitamos para calcular la calificación promedio de la aplicación usando un bucle `for`.

- Calcula la calificación promedio de las aplicaciones almacenadas en la variable `app_data_set`.

    - Inicializa una variable denominada `rating_sum` con un valor de cero fuera del cuerpo del bucle.
    - Recorre (itera) la lista de listas `app_data_set`. Para cada una de las cinco iteraciones del bucle (para cada fila en `app_data_set`):
        - Extrae la calificación de la aplicación y guárdala en una variable denominada `rating`. La calificación es el último elemento de cada fila.
        - Agrega el valor almacenado en `rating` al valor actual de `rating_sum` y asigna el resultado nuevamente a `rating_sum`.
        - Fuera del cuerpo del bucle `for`, divide la suma de calificaciones (almacenada en `rating_sum`) por el número de calificaciones para obtener un valor promedio. Guarda el resultado en una variable denominada `avg_rating` e imprimela.

Escribe tu codigo abajo de esta sección ⏬

---

In [ ]:
row_1 = ['Facebook', 0.0, 'USD', 2974676, 3.5]
row_2 = ['Instagram', 0.0, 'USD', 2161558, 4.5]
row_3 = ['Clash of Clans', 0.0, 'USD', 2130805, 4.5]
row_4 = ['Fruit Ninja Classic', 1.99, 'USD', 698516, 4.5]
row_5 = ['Minecraft: Pocket Edition', 6.99, 'USD', 522012, 4.5]

app_data_set = [row_1, row_2, row_3, row_4, row_5]

# Escribe tu codigo debajo



Escribe tu codigo arriba de esta sección ⏫

---

<a id="Abrir-un-archivo"></a>
## 5. Abrir un archivo
[[Contenido]](#Contenido) 
[[Ejercicio de esta sección]](#Instrucciones-Abrir-un-archivo)

Ahora aprendamos a leer un conjunto de datos grande como una lista de listas, para que podamos calcular la calificación promedio de la aplicación.

El conjunto de datos que vamos a utilizar tiene **7197 filas** y **17 columnas**, lo que equivale a 122,349 puntos de datos ($7197 \cdot 17$). Escribir todo esto nos llevaría días. También seria muy probable que cometamos errores tipográficos, lo que podría dar lugar a datos erróneos y conclusiones falsas. Afortunadamente, podemos aprovechar Python para almacenar este conjunto de datos como una lista de listas en cuestión de segundos.

Un conjunto de datos generalmente se almacena como un archivo en una computadora; el conjunto de datos que vamos a usar se almacena como un archivo llamado `AppleStore.csv`, donde la extension `csv` significa *valores separados por coma o en ingles comma separated values*. Comenzamos abriendo el archivo usando la función `open()`:

In [2]:
# Mobile App Store Dataset (Ramanathan Perumal)
opened_file = open('/kaggle/input/app-store-apple-data-set-10k-apps/AppleStore.csv')
print(opened_file)

<_io.TextIOWrapper name='/kaggle/input/app-store-apple-data-set-10k-apps/AppleStore.csv' mode='r' encoding='UTF-8'>


La cadena de caracteres que se le pasa a la funcion `open` debe ser la ruta al archivo `AppleStore.csv`, por ejemplo,
- en el caso de `Kaggle` la ruta es: `/kaggle/input/app-store-apple-data-set-10k-apps/AppleStore.csv`
- en el caso de que el archivo este en la misma carpeta que el archivo de este notebook la ruta es: `AppleStore.csv`.

La salida de la celda de codigo anterior nos indica que lo que regresa la función `open()` es un **objeto**. Una vez que hemos abierto el archivo con `open()`, lo leemos usando una función `reader()`. Importamos la función `reader()` del módulo `csv` usando el código `from csv import reader`

In [2]:
opened_file = open('/kaggle/input/app-store-apple-data-set-10k-apps/AppleStore.csv')

from csv import reader
read_file = reader(opened_file)
print(read_file)

La salida de la celda de codigo anterior nos indica que lo que regresa la función `reader()` es un objeto. Ahora que hemos leído el archivo, podemos transformarlo en una lista de listas usando la función `list()`:

In [4]:
opened_file = open('/kaggle/input/app-store-apple-data-set-10k-apps/AppleStore.csv')

from csv import reader
read_file = reader(opened_file)
apps_data = list(read_file)

La variable `apps_data` anterior es una lista de listas y almacena un conjunto de datos de 7197 filas y 17 columnas. A continuación, imprimimos solo las primeras cinco filas de `apps_data`

In [35]:
opened_file = open('/kaggle/input/app-store-apple-data-set-10k-apps/AppleStore.csv')

from csv import reader
read_file = reader(opened_file)
apps_data = list(read_file)

print(apps_data[:5])

[['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic'], ['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1'], ['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1'], ['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1'], ['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1']]


La salida de datos de la celda anterior no permite ver con claridad donde termina una lista y donde empieza otra. A continuacion usamos colores sobre la salida de texto del codigo de arriba para poder identificar mejor cada una de las listas.

<p>[<span style="color: blue;">['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']</span>, <span style="color: red;">['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']</span>, <span style="color: green;">['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']</span>, <span style="color: magenta;">['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']</span>, <span style="color: orange;">['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1']</span>]</p>

Aunque hay **7197** filas (aplicaciones) en nuestro conjunto de datos, `len(apps_data)` indica que hay 7198 filas porque también considera la fila del encabezado, que describe los nombres de las columnas (la primera fila).

In [36]:
opened_file = open('/kaggle/input/app-store-apple-data-set-10k-apps/AppleStore.csv')

from csv import reader
read_file = reader(opened_file)
apps_data = list(read_file)

print(len(apps_data))

7198


<a id="Instrucciones-Abrir-un-archivo"></a>
## Instrucciones
[[Contenido]](#Contenido) 
[[Teoria de esta sección]](#Abrir-un-archivo)

Explora `apps_data` y haz lo siguiente:
- Imprime la longitud de `apps_data` usando la función len().
- Imprime el numero de columnas de `apps_data` usando la función len().
- Imprime la primera fila (la fila de los nombres de las columnas).
- Imprime el encabezado de la columna 9.
- Imprime la segunda y tercera fila.

Escribe tu codigo abajo de esta sección ⏬

---

In [2]:
opened_file = open('/kaggle/input/app-store-apple-data-set-10k-apps/AppleStore.csv')

from csv import reader
read_file = reader(opened_file)
apps_data = list(read_file)

# Escribe tu codigo debajo



Escribe tu codigo arriba de esta sección ⏫

---

<a id="Calcular-promedio"></a>
## 6. Calcular promedio
[[Contenido]](#Contenido) 
[[Ejercicio de esta sección]](#Instrucciones-Calcular-promedio)

Ahora lo que queremos hacer es calcular el promedio de la columna 9 del conjunto de datos (`user_rating`), para realizar esta operacion debemos de observar 2 cosas:
1. La primera fila del conjunto de datos `apps_data[0]`, contiene los encabezados del conjunto de datos:

In [15]:
print("Encabezados:")
print(apps_data[0])
print("\nColumna 9:")
print(apps_data[0][8])

Encabezados:
['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']

Columna 9:
user_rating


2. Cada uno de los valores de las filas de datos (i.e. filas 1-7198) son `string`:

In [16]:
for encabezado in apps_data[1]: # Fila 1
    print(type(encabezado), "valor: ", encabezado)

<class 'str'> valor:  1
<class 'str'> valor:  281656475
<class 'str'> valor:  PAC-MAN Premium
<class 'str'> valor:  100788224
<class 'str'> valor:  USD
<class 'str'> valor:  3.99
<class 'str'> valor:  21292
<class 'str'> valor:  26
<class 'str'> valor:  4
<class 'str'> valor:  4.5
<class 'str'> valor:  6.3.5
<class 'str'> valor:  4+
<class 'str'> valor:  Games
<class 'str'> valor:  38
<class 'str'> valor:  5
<class 'str'> valor:  10
<class 'str'> valor:  1


<a id="Instrucciones-Calcular-promedio"></a>
## Instrucciones
[[Contenido]](#Contenido) 
[[Teoria de esta sección]](#Calcular-promedio)

Calcule la calificación promedio de las aplicaciones para las 7,197 aplicaciones almacenadas en el conjunto de datos.

- Inicializa una variable denominada `rating_sum` con un valor de cero.
- Recorre la lista de listas `apps_data[1:]` (asegúrese de no incluir la fila del encabezado). Para cada una de las 7197 iteraciones del bucle (para cada fila en `apps_data[1:]`), haz lo siguiente:
    - Extrae la calificación de la aplicación y guárdala en una variable denominada `rating` (la calificación tiene el número de índice 8). Convierte el valor de `rating` de una cadena a un flotante usando la función `float()`.
    - Agrega el valor almacenado en `rating` al valor actual de `rating_sum` y asigna el resultado nuevamente a `rating_sum`.
- Divide la suma de las calificaciones (almacenada en `rating_sum`) por la cantidad de calificaciones para obtener un valor promedio. Guarda el resultado en una variable denominada `avg_rating` e imprimela.

Escribe tu codigo abajo de esta sección ⏬

---

In [2]:
opened_file = open('/kaggle/input/app-store-apple-data-set-10k-apps/AppleStore.csv')

from csv import reader
read_file = reader(opened_file)
apps_data = list(read_file)

# Escribe tu codigo debajo



Escribe tu codigo arriba de esta sección ⏫

---

<a id="Append-calcular-promedio"></a>
## 7. Funcion `append()` para calcular un promedio
[[Contenido]](#Contenido) 
[[Ejercicio de esta sección]](#Instrucciones-Append-calcular-promedio)

Ahora aprenderemos un método alternativo para calcular el valor de calificación promedio. Una vez que creamos una lista, podemos agregarle valores usando la función `append()`:

In [17]:
a_list = [1, 2]
a_list.append(3)
print(a_list)

[1, 2, 3]


A continuación se explica cómo agregar valores a una lista vacía:

In [18]:
empty_list = []
empty_list.append(12)
print(empty_list)

[12]


<a id="Instrucciones-Append-calcular-promedio"></a>
## Instrucciones
[[Contenido]](#Contenido) 
[[Teoria de esta sección]](#Append-calcular-promedio)

Utilizando la nueva técnica que hemos aprendido, calcula la calificación promedio de las aplicaciones para las 7,197 aplicaciones almacenadas en nuestro conjunto de datos.

- Inicializa una lista vacía llamada `rating_sum`.
- Recorre la lista de listas `apps_data[1:]` (no incluyas la fila del encabezado). Para cada una de las 7197 iteraciones del ciclo, haz lo siguiente:
    - Extrae la calificación de la aplicación y guárdala en una variable denominada `rating` (la calificación tiene el número de índice 8). Convierte el valor de `rating` de una cadena a un flotante.
    - Agrega el valor almacenado en `rating` a la lista `rating_sum`.
- Calcule la suma de todas las calificaciones usando la función `sum()`.
- Divide la suma de todas las calificaciones por el número de calificaciones y asigna el resultado a una variable denominada `avg_rating`.
- Imprime los resultados de `avg_rating`.

Escribe tu codigo abajo de esta sección ⏬

---

In [2]:
opened_file = open('/kaggle/input/app-store-apple-data-set-10k-apps/AppleStore.csv')

from csv import reader
read_file = reader(opened_file)
apps_data = list(read_file)

# Escribe tu codigo debajo



Escribe tu codigo arriba de esta sección ⏫

---